src/analytics/AnalyticsManager.ts

```diff
@@ -0,0 +1,497 @@
+ * - Users can opt-out via analytics.enabled = false
+ *
+ * Override:
+ * Set NEXUS_POSTHOG_KEY env var or pass custom key in SDK config
+ */
+const DEFAULT_POSTHOG_KEY = process.env.NEXUS_POSTHOG_KEY || 'phc_UD6lQU3PEw1d8oo8E17rJLmRAR7kxJbQ5OseHuCvi7N';
+const DEFAULT_POSTHOG_HOST = 'https://us.i.posthog.com';
+
+export class AnalyticsManager {
+  private provider: AnalyticsProvider;
+  private session: SessionManager;
+  private performance: PerformanceTracker;
+  private config: AnalyticsConfig;
+  private enabled: boolean;
+  private sdkVersion: string;
+  private network?: 'mainnet' | 'testnet';
+
+  constructor(config?: AnalyticsConfig, sdkVersion: string = '0.0.1', network?: 'mainnet' | 'testnet') {
+    this.config = config || { enabled: true };
+    this.enabled = config?.enabled !== false; // Default: true (opt-out)
+    this.sdkVersion = sdkVersion;
+    this.network = network;
+
+    // Initialize session and performance trackers
+    this.session = new SessionManager();
+    this.performance = new PerformanceTracker();
+
+    // Initialize the appropriate provider
+    if (this.enabled) {
+      const posthogApiKey = config?.posthogApiKey || DEFAULT_POSTHOG_KEY;
+      const posthogApiHost = config?.posthogApiHost || DEFAULT_POSTHOG_HOST;
+
+      this.provider = new PostHogProvider({
+        apiKey: posthogApiKey,
+        apiHost: posthogApiHost,
+        sessionRecording: config?.sessionRecording || false,
+        debug: config?.debug || false,
+      });
+    } else {
+      this.provider = new NoOpProvider();
+    }
+
+    // Initialize the provider
+    this.provider.initialize();
+
+    // Register global properties
+    if (this.enabled && this.provider.isInitialized()) {
+      this.registerGlobalProperties();
+    }
+
+    // Track session start
+    this.trackSessionStart();
+  }
+
+  /**
+   * Register global properties that are sent with every event
+   */
+  private registerGlobalProperties(): void {
+    const globalProps: Record<string, unknown> = {
+      sdkVersion: this.sdkVersion,
+      sessionId: this.session.getSessionId(),
+    };
+
+    if (this.network) {
+      globalProps.network = this.network;
+    }
+
+    if (this.config.appMetadata?.appName) {
+      globalProps.appName = this.config.appMetadata.appName;
+    }
+
+    if (this.config.appMetadata?.appVersion) {
+      globalProps.appVersion = this.config.appMetadata.appVersion;
+    }
+
+    if (this.config.appMetadata?.appUrl) {
+      globalProps.appUrl = this.config.appMetadata.appUrl;
+    }
+
+    // Add environment if in browser
+    if (typeof window !== 'undefined') {
+      globalProps.environment = this.detectEnvironment();
+    }
+
+    this.provider.register(globalProps);
+
+    if (this.config.debug) {
+      console.log('[AnalyticsManager] Global properties registered:', globalProps);
+    }
+  }
+
+  /**
+   * Detect environment (development/production)
+   */
+  private detectEnvironment(): string {
```

Comment:
Aren't `server`,  and `development` /  `production`  different things?
One is a type of platform (`server`/`browser`) and the other one is is it either a local or non local env?